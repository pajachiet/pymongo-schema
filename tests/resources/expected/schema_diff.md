
### Database: db0
|Hierarchy                            |Previous Schema              |New Schema                    |
|-------------------------------------|-----------------------------|------------------------------|
|                                     |db0                          |None                          |


### Database: db1
|Hierarchy                            |Previous Schema              |New Schema                    |
|-------------------------------------|-----------------------------|------------------------------|
|                                     |None                         |db1                           |


### Database: db
#### Collection: coll1 
|Hierarchy                            |Previous Schema              |New Schema                    |
|-------------------------------------|-----------------------------|------------------------------|
|                                     |coll1                        |None                          |

#### Collection: coll2 
|Hierarchy                            |Previous Schema              |New Schema                    |
|-------------------------------------|-----------------------------|------------------------------|
|                                     |None                         |coll2                         |

#### Collection: coll 
|Hierarchy                            |Previous Schema              |New Schema                    |
|-------------------------------------|-----------------------------|------------------------------|
|                                     |field2                       |None                          |
|                                     |None                         |field4                        |
|field3                               |{"type": "boolean"}          |{"type": "string"}            |
|field.array_subfield                 |None                         |subsubfield2                  |
|field.array_subfield.subsubfield     |{"type": "integer"}          |{"type": "boolean"}           |
|field5                               |{"array_type": "string"}     |{"array_type": "integer"}     |
|field6                               |{"type": "ARRAY"}            |{"type": "string"}            |

