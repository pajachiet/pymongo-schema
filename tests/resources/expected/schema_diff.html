<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>DataDict</title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>

    <style>
    table {
        font-family: arial, sans-serif;
        border-collapse: collapse;
        width: 100%;
    }

    td, th {
        border: 1px solid #dddddd;
        text-align: left;
        padding: 8px;
    }

    tr:nth-child(even) {
        background-color: #F0F8FF;
    }

    .level1 {
        background-color: lightblue !important;
    }
    </style>
</head>
<body>

<div class="panel-group">
    
    <div class="panel level1">
        <div class="panel-heading">
            <h4>
                <a data-toggle="collapse" href="#collapse_db0" style="color: #333333 !important;">db0</a>
            </h4>
        </div>
        <div id="collapse_db0" class="panel-collapse collapse">
            <div class="panel-group" >
                
                <div class="panel panel-default">
                    
                    <div id="collapse_db0_" class="panel-collapse">
                    
                        <div class="panel-body">
                            <table>
                                <tr>
                                    
                                    <th style="text-align: left">Hierarchy</th>
                                    
                                    <th style="text-align: left">Previous Schema</th>
                                    
                                    <th style="text-align: left">New Schema</th>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td></td>
                                    
                                    <td>db0</td>
                                    
                                    <td>None</td>
                                    
                                </tr>
                                
                            </table>
                        </div>
                    </div>
                </div>
                
            </div>
        </div>
    </div>
    
    <div class="panel level1">
        <div class="panel-heading">
            <h4>
                <a data-toggle="collapse" href="#collapse_db1" style="color: #333333 !important;">db1</a>
            </h4>
        </div>
        <div id="collapse_db1" class="panel-collapse collapse">
            <div class="panel-group" >
                
                <div class="panel panel-default">
                    
                    <div id="collapse_db1_" class="panel-collapse">
                    
                        <div class="panel-body">
                            <table>
                                <tr>
                                    
                                    <th style="text-align: left">Hierarchy</th>
                                    
                                    <th style="text-align: left">Previous Schema</th>
                                    
                                    <th style="text-align: left">New Schema</th>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td></td>
                                    
                                    <td>None</td>
                                    
                                    <td>db1</td>
                                    
                                </tr>
                                
                            </table>
                        </div>
                    </div>
                </div>
                
            </div>
        </div>
    </div>
    
    <div class="panel level1">
        <div class="panel-heading">
            <h4>
                <a data-toggle="collapse" href="#collapse_db" style="color: #333333 !important;">db</a>
            </h4>
        </div>
        <div id="collapse_db" class="panel-collapse collapse">
            <div class="panel-group" >
                
                <div class="panel panel-default">
                    
                    <div class="panel-heading">
                        <h4 class="panel-title">
                            <a data-toggle="collapse" href="#collapse_db_coll1">coll1</a>
                        </h4>
                    </div>
                    <div id="collapse_db_coll1" class="panel-collapse collapse">
                    
                        <div class="panel-body">
                            <table>
                                <tr>
                                    
                                    <th style="text-align: left">Hierarchy</th>
                                    
                                    <th style="text-align: left">Previous Schema</th>
                                    
                                    <th style="text-align: left">New Schema</th>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td></td>
                                    
                                    <td>coll1</td>
                                    
                                    <td>None</td>
                                    
                                </tr>
                                
                            </table>
                        </div>
                    </div>
                </div>
                
                <div class="panel panel-default">
                    
                    <div class="panel-heading">
                        <h4 class="panel-title">
                            <a data-toggle="collapse" href="#collapse_db_coll2">coll2</a>
                        </h4>
                    </div>
                    <div id="collapse_db_coll2" class="panel-collapse collapse">
                    
                        <div class="panel-body">
                            <table>
                                <tr>
                                    
                                    <th style="text-align: left">Hierarchy</th>
                                    
                                    <th style="text-align: left">Previous Schema</th>
                                    
                                    <th style="text-align: left">New Schema</th>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td></td>
                                    
                                    <td>None</td>
                                    
                                    <td>coll2</td>
                                    
                                </tr>
                                
                            </table>
                        </div>
                    </div>
                </div>
                
                <div class="panel panel-default">
                    
                    <div class="panel-heading">
                        <h4 class="panel-title">
                            <a data-toggle="collapse" href="#collapse_db_coll">coll</a>
                        </h4>
                    </div>
                    <div id="collapse_db_coll" class="panel-collapse collapse">
                    
                        <div class="panel-body">
                            <table>
                                <tr>
                                    
                                    <th style="text-align: left">Hierarchy</th>
                                    
                                    <th style="text-align: left">Previous Schema</th>
                                    
                                    <th style="text-align: left">New Schema</th>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td></td>
                                    
                                    <td>field2</td>
                                    
                                    <td>None</td>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td></td>
                                    
                                    <td>None</td>
                                    
                                    <td>field4</td>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td>field3</td>
                                    
                                    <td>{'type': 'boolean'}</td>
                                    
                                    <td>{'type': 'string'}</td>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td>field.array_subfield</td>
                                    
                                    <td>None</td>
                                    
                                    <td>subsubfield2</td>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td>field.array_subfield.subsubfield</td>
                                    
                                    <td>{'type': 'integer'}</td>
                                    
                                    <td>{'type': 'boolean'}</td>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td>field5</td>
                                    
                                    <td>{'array_type': 'string'}</td>
                                    
                                    <td>{'array_type': 'integer'}</td>
                                    
                                </tr>
                                
                                <tr>
                                    
                                    <td>field6</td>
                                    
                                    <td>{'type': 'ARRAY'}</td>
                                    
                                    <td>{'type': 'string'}</td>
                                    
                                </tr>
                                
                            </table>
                        </div>
                    </div>
                </div>
                
            </div>
        </div>
    </div>
    
</div>

</body>
</html>